Plutarch/Rec/TH.hs

```diff
@@ -37,10 +41,20 @@ bare :: TH.Type -> Q TH.Type
 bare (TH.SigT t _) = bare t
 bare t = pure t
 
-reifyConstructor :: TH.Name -> Q TH.Con
+#if MIN_VERSION_template_haskell(2,17,0)
+bindingName :: TH.TyVarBndr flag -> TH.Name
+bindingName (TH.PlainTV name _) = name
+bindingName (TH.KindedTV name _ _) = name
+#else
+bindingName :: TH.TyVarBndr -> TH.Name
+bindingName (TH.PlainTV name) = name
+bindingName (TH.KindedTV name _) = name
+#endif
+
+reifyConstructor :: TH.Name -> Q (TH.Con, [TH.Name])
 reifyConstructor ty = do
   (TH.TyConI tyCon) <- TH.reify ty
   case tyCon of
-    TH.DataD _ _nm _tyVars _kind [c] _ -> return c
-    TH.NewtypeD _ _nm _tyVars _kind c _ -> return c
+    TH.DataD _ _nm tyVars _kind [c] _ -> return (c, bindingName <$> tyVars)
```

Comment:
Separate PR?

Reply:
Okay, I'll cherry-pick this into a separate PR.

Reply:
I forgot I already put this in the merged PR #453, I just need to rebase this PR on top of staging.